clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -756,12 +756,11 @@ protected boolean onJoinPrepare(Timer timer, int generation, String memberId) {
             joinPrepareTimer.update();
         }
 
-        final SortedSet<TopicPartition> partitionsToRevoke = getPartitionsToRevoke(protocol, generation, memberId);
-
+        final SortedSet<TopicPartition> eagerPartitionsToRevoke = eagerPartitionsToRevoke(protocol);
```

Comment:
Hmm, thinking about this a little more since we're down to just the eager protocol. Since the assignment won't change until the rebalance completes, maybe we do not need to precompute it. In other words, maybe we can restore the original logic in `revokePartitions` and we can change `markPendingPartitions` to something like this:

```java
private void maybeMarkPartitionsPendingRevocation() {
  if (protocol == RebalanceProtocol.EAGER) {
    // When asynchronously committing offsets prior to the revocation of a set of partitions, there will be a
    // window of time between when the offset commit is sent and when it returns and revocation completes. It is
    // possible for pending fetches for these partitions to return during this time, which means the application's
    // position may get ahead of the committed position prior to revocation. This can cause duplicate consumption.
    // To prevent this, we mark the partitions as "pending revocation," which stops the Fetcher from sending new
    // fetches or returning data from previous fetches to the user.
    Set<TopicPartition> partitions = subscriptions.assignedPartitions()
    log.debug("Marking assigned partitions pending for revocation: {}", partitions);
    subscriptions.markPendingRevocation(partitions);
  }
}
```